### ISSUE TITLE
Reimplement overlays filtering system

### ISSUE THREAD

**Issue by johnd0e** [IITC-CE/ingress-intel-total-conversion#490]
To provide filtering **by faction** iitc creates 3 separate groups (for each portal level / links / fields).
To provide filtering portals **by level** iitc creates 9 separate groups.
Pretty reliable system, with [minimal](https://github.com/IITC-CE/ingress-intel-total-conversion/blob/938da84bc721998cfb0fef9c2e7c75d117a3b1cc/core/code/map_data_render.js#L530-L538) overhead.

* Unfortunately it is limited to standard overlays (e.g. Player Tracker / Zaprange unable to benefit from it, and have to create separate 'factionated' overlays).
* and not flexible enough to handle complex cases, such as described in #197
  - ornaments are shown unconditionally, even for hidden portals.
  - no possibility to show only portals with ornaments
  - etc

**Comment by johnd0e** [comment]
I have some ideas how to implement that.

1) Current factions/levels groups can be replaced with filters, visually represented exactly like now - as items in layer chooser.
But if fact they will provide functions to filter any entities list.
2) Every entity should be able to participate in arbitrary number of groups (not only one like now), and entity should stay on map as long as it is still present in any group (could be implemented by storing groups list in entity itself)
3) Some groups should be assigned to have priority over filters.

**Comment by le-jeu** [comment]
~~Related: #457~~

**Comment by johnd0e** [comment]
@le-jeu not really: there are discussed portallist-dialog-only features limited to presentation of table items, but here we are talking about layerChooser overlays, controlling entities on the map.
There is literally nothing in common.

**Comment by le-jeu** [comment]
Mistype for #451
